### ISSUE TITLE
More warnings with js-controller 4.0-alpha

### ISSUE THREAD

**Issue by Grothesk242** [DrozmotiX/ioBroker.tado#459]
New js-controller iobroker.js-controller@4.0.0-alpha.67-20220131-6621d47f

throws many many new warnings: 

```2022-02-01 19:19:54.080  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:54.080  - warn: tado.0 (2589) Object 109324.Rooms.1.nextScheduleChange.setting.temperature.fahrenheit is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:54.080  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:54.531  - warn: tado.0 (2589) Object 109324.Rooms.3.nextScheduleChange.start is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:54.531  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:54.532  - warn: tado.0 (2589) Object 109324.Rooms.3.nextScheduleChange.setting.type is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:54.532  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:54.533  - warn: tado.0 (2589) Object 109324.Rooms.3.nextScheduleChange.setting.temperature.celsius is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:54.533  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:54.534  - warn: tado.0 (2589) Object 109324.Rooms.3.nextScheduleChange.setting.temperature.fahrenheit is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:54.534  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:55.001  - warn: tado.0 (2589) Object 109324.Rooms.2.nextScheduleChange.start is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:55.002  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:55.002  - warn: tado.0 (2589) Object 109324.Rooms.2.nextScheduleChange.setting.type is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:55.003  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:55.004  - warn: tado.0 (2589) Object 109324.Rooms.2.nextScheduleChange.setting.temperature.celsius is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:55.004  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:55.004  - warn: tado.0 (2589) Object 109324.Rooms.2.nextScheduleChange.setting.temperature.fahrenheit is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:55.005  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:55.468  - warn: tado.0 (2589) Object 109324.Rooms.4.nextScheduleChange.start is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:55.468  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:55.469  - warn: tado.0 (2589) Object 109324.Rooms.4.nextScheduleChange.setting.type is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:55.469  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:55.470  - warn: tado.0 (2589) Object 109324.Rooms.4.nextScheduleChange.setting.temperature.celsius is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:55.470  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.
2022-02-01 19:19:55.471  - warn: tado.0 (2589) Object 109324.Rooms.4.nextScheduleChange.setting.temperature.fahrenheit is invalid: obj.common.states has an invalid type! Expected "object", received "string"
2022-02-01 19:19:55.471  - warn: tado.0 (2589) This object will not be created in future versions. Please report this to the developer.```

**Comment by HGlab01** [comment]
Hi, thanks for raising the issue.
You are right, a lot of issues, but I am not sure if this is an adapter issue.

Example:
I checked Rooms.x.nextScheduleChange.setting.temperature.fahrenheit
and it is defined as number, the API sends a number. So neither "object" nor "string" makes sense in this log.
{
  "type": "state",
  "common": {
    "modify": "",
    "name": "Temperature Fahrenheit",
    "type": "number",
    "role": "value.temperature",
    "read": true,
    "unit": "°F",
    "states": ""
 }

As you used an alpha version I guess it is a controller issue. Which adapter version did you use?

**Comment by HGlab01** [comment]
Please do me a flavor and 
1. go to objects
2. toggle expert mode
3. select one object where you have a log line
4. hit the "pencil" in this line on right hand side

You will see something like this

```
{
  "type": "state",
  "common": {
    "modify": "",
    "name": "Temperature Fahrenheit",
    "type": "number",
    "role": "value.temperature",
    "read": true,
    "unit": "°F",
    "write": true,
    "states": ""
  },
  "from": "system.adapter.tado.0",
  "user": "system.user.admin",
  "ts": 1643354406185,
  "_id": "tado.0.xxxxxx.Rooms.1.nextScheduleChange.setting.temperature.fahrenheit",
  "acl": {
    "object": 1632,
    "state": 1632,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator"
  }
}
```

Please change 
`"states": ""`
to
`"states": null`

Then press "Write" and restart the adapter.
Is the log line gone for the specific state?

**Comment by Grothesk242** [comment]
Setting null manually did not change the behaviour. Log is still flooded
with messages.

HGlab ***@***.***> schrieb am Di., 1. Feb. 2022, 20:45:

**Comment by HGlab01** [comment]
Fixed in 0.3.15-alpha.0 (https://github.com/DrozmotiX/ioBroker.tado/releases/tag/v0.3.15-alpha.0) in combination with the next js-controller v4 alpha-release

**Comment by Grothesk242** [comment]
Adapterstart is now silent: 

```2022-02-02 21:13:05.554  - info: host.chet instance system.adapter.tado.0 started with pid 50644
2022-02-02 21:13:07.693  - info: tado.0 (50644) starting. Version 0.3.15-alpha.0 (non-npm: DrozmotiX/ioBroker.tado) in /opt/iobroker/node_modules/iobroker.tado, node: v16.13.2, js-controller: 4.0.0-alpha.70-20220202-ee665b37
2022-02-02 21:13:07.727  - info: tado.0 (50644) Started with JSON-Explorer version 0.1.9-alpha.2
2022-02-02 21:13:08.336  - info: tado.0 (50644) Connected to Tado cloud, initialyzing...
2022-02-02 21:13:13.198  - info: tado.0 (50644) Initialisation finished, connected to Tado cloud service refreshing every 30 seconds```

**Comment by HGlab01** [comment]
Fixed in 0.3.15-alpha.0
https://github.com/DrozmotiX/ioBroker.tado/releases/tag/v0.3.15-alpha.0
